tests/Aspire.Hosting.Testing.Tests/TestingBuilderTests.cs

```diff
@@ -13,6 +13,34 @@ namespace Aspire.Hosting.Testing.Tests;
 
 public class TestingBuilderTests
 {
+    [Theory]
+    [RequiresDocker]
+    [InlineData(false)]
+    [InlineData(true)]
+    public async Task CreateAsyncWithOptions(bool genericEntryPoint)
+    {
+        var nonExistantRegistry = "non-existant-registry-azurecr.io";
+        Action<DistributedApplicationOptions, HostApplicationBuilderSettings> configureBuilder = (options, settings) =>
+        {
+            options.ContainerRegistryOverride = nonExistantRegistry;
```

Comment:
How can I test setting something on `settings`?

Reply:
Set the `settings.EnvironmentName` to something like `Production` or `Development` or `CustomEnvironment`.

And then verify that `appHost.Environment.EnvironmentName` is set to that name.